### PR TITLE
Refactored CI and Tests

### LINKED ISSUE #2928: Rewrite the test cases that are failing sometimes in our CI
**Context**
Sometimes our CI is failing due to faulty test cases, particularly instrument tests. We should rewrite/fix them. <del>For example, our [IntroRobot](https://github.com/kiwix/kiwix-android/blob/develop/app/src/androidTest/java/org/kiwix/kiwixmobile/intro/IntroRobot.kt) is hugely unreliable to our CI</del>.
**Expected Behaviour**
A green CI :)

### PR DESCRIPTION
**Pull request by gouri-panda**
Fixes #2928 
**This PR Changes**
* Added macOS version 11 in coverage and nightly
* Added Pixel 2 in Profile 
* Added Google API 21 & 30
* Refactor the tests that are most likely to fail in CI sometimes

### CODE REVIEW

**Comment by kelson42** [comment]
Blocked by #2958

**Comment by kelson42** [comment]
@gouri-panda Can we consider we merge this and benefit of all other improvments and we just keep https://github.com/kiwix/kiwix-android/issues/2958 open for the moment?

**Comment by gouri-panda** [comment]
@kelson42 #2958 blocking this.

**Comment by gouri-panda** [comment]
@MohitMaliFtechiz It's ready now :)

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -15,9 +15,9 @@ jobs:
   coverageReport:
     strategy:
       matrix:
-        api-level: [21,30]
```
Comment (MohitMaliFtechiz) [inline .github/workflows/coverage.yml:18]: hi  @gouri-panda , why we are removing api level 21 ?

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/kiwix/kiwixmobile/page/history/HistoryRobot.kt [hunk 2]
```diff
@@ -28,9 +27,6 @@ import org.kiwix.kiwixmobile.R
 fun history(func: HistoryRobot.() -> Unit) = HistoryRobot().applyWithViewHierarchyPrinting(func)
 
 class HistoryRobot : BaseRobot() {
-  init {
-    assertDisplayed(R.string.history_from_current_book)
```
Comment (MohitMaliFtechiz) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/page/history/HistoryRobot.kt:32]: hi @gouri-panda,

Why we are removing init block  ?

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (gouri-panda) [reply .github/workflows/coverage.yml:18]: It's unnecessary now. Now we are testing this nightly, there no need to be double-checked.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/page/history/HistoryRobot.kt (hunk 2 above)
Comment (gouri-panda) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/page/history/HistoryRobot.kt:32]: Often times `BaseRobot` class doesn't instantiate before it's checking the `assertDisplayed()` function.

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (MohitMaliFtechiz) [reply .github/workflows/coverage.yml:18]: hi @kelson42 can you please confirm this ?

Comment (kelson42) [reply .github/workflows/coverage.yml:18]: i have no clue about the consequences and impact of such a decision. If you want me to take a decision, please let me know clearly what it implies.

Comment (gouri-panda) [reply .github/workflows/coverage.yml:18]: @kelson42 Most of the project happy with 1 API(we did that before with only API 21) . We can test all the different APIs in nightly.

**Review by MohitMaliFtechiz — DISMISSED** [review]

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (kelson42) [reply .github/workflows/coverage.yml:18]: Api21 is the older API right? So this is corresponding to the oldes version of A ndroid we support i guess. How to you check that we don’t inadvertantly stop to support old versions of An droid devices if you remove CI with old API?

Comment (gouri-panda) [reply .github/workflows/coverage.yml:18]: @kelson42 We are checking with the nightly.

**Review by kelson42 — CHANGES_REQUESTED** [review]
I'm not confortable for the moment to remove API21 in CI without clear explanation.

**Review by kelson42 — CHANGES_REQUESTED** [review]
Here my remarks:
* The PR has no description
* We should use clearly identify OSes, so no `latest`
* It seems the ticket #2928 is not fully fixed by this PR, so we should not write `Fixes #2928`
* Actually #2928 talks about revamping test, here there is no real test code change
*

**Review by MohitMaliFtechiz — APPROVED** [review]

**Pull request merged**